src/shared/components/RichTable.tsx

```diff
@@ -616,12 +646,14 @@ const useRichTable = <D extends object = {}>({
 const RichTablePagination = ({ bag }: { bag: RichTableBag<any> }) => {
   const { page, numPages, setPage, reqInfo, executeAction } = bag;
   const [newPage, setNewPage] = useState(page.toString());
-  // we need a ref to store the new page to get around the fake blur listener capturing outdated values of newPage
-  const newPageRef = useRef(page.toString());
-  const hasBlurListener = useRef(false);
+  // we need a ref to store the page to get around the fake blur listener capturing outdated values of page
+  const pageRef = useRef(page);
+  const fakeBlurListener = useRef<((e: MouseEvent) => void) | null>(null);
```

Comment:
Why are we using a fake blur listener here? Whatever the reason, it should probably be documented.

Reply:
Documentation added: 933e930